### PR TITLE
feat: SQL Server export SP leaves scalar and complex answer fields empty in default CSV

### LINKED ISSUE #891: SQL Server export SP leaves scalar and complex answer fields empty in default CSV
### Goal

Fix SQL Server `export_form_submissions` - latest version, so default CSV export includes submitted scalar and complex answer values (not empty cells), and lock the gap with integration coverage based on the AllQuestions fixture until full MS SQL Server export support is in place.

### Steps to Reproduce

1. Run Endatix against **SQL Server** with the current stored procedure (e.g. `/src/Endatix.Persistence.SqlServer/Scripts/Procedures/export_form_submissions_v3.sql` / current `v4` equivalent).
2. Create a form from `/tests/Endatix.Modules.Reporting.Tests/Features/FormSchema/FlattenedFormDefinition/Fixtures/AllQuestions/all-questions-definition.json` and submit answers covering scalar and complex question types.
3. Export submissions via the default CSV path (SQL data source + standard tabular exporter).

### Actual

CSV columns for many answer fields are empty (`""` / blank). Scalar values are dropped; complex/structured answers are also missing or empty instead of the submitted JSON/values.

Likely cause in the SP answer projection: values are taken with `JSON_QUERY` only, which returns `NULL` for scalars; `ISNULL(..., '""')` then writes empty strings into `AnswersJson`:

```sql
ISNULL(JSON_QUERY(r.JsonData, @path), '""')
```

### Expected

Each exported answer column contains the submitted value:
- scalars via `JSON_VALUE` (or equivalent)
- objects/arrays via `JSON_QUERY` (preserving JSON structure)

Parity with PostgreSQL export behavior for the same AllQuestions fixture.

### Implementation Notes

- **Verify missing fields** against AllQuestions:
  - Definition: `.../Fixtures/AllQuestions/all-questions-definition.json`
  - Use matching submission fixture(s) under the same folder where available
  - Compare exported keys/values to expected export coverage used elsewhere (e.g. Reporting tabular export tests that load AllQuestions via `FormSchemaFixtureLoader`)
- **Integration tests (SQL Server):**
  - Add (or extend) SQL Server export integration tests that assert non-empty answer cells for representative scalar + complex question types from AllQuestions
  - Keep tests failing or explicitly documenting the gap until a new SQL Server export SP / data-source path fully supports this projection
  - Prefer immutable SP versioning (`vN+1`) — do not edit shipped scripts in place
- Scope is **Endatix API / persistence** (`endatix-api`), not Hub UI.

### PR DESCRIPTION
**Pull request by PeterMarinov**
# feat(e891): SQL Server export SP leaves scalar and complex answer fields empty in default CSV

## Description

>[!NOTE]
>SInce the `SoftDeleteSubmissions` migration is still not merged to main, we are modifying the `export_form_submissions_v4.sql` directly. This is why we have chosen to dely merging the `submission-and-export-changes` into `main`

- Fixing error during setup due to short signing key (default value) and misleading error
- Applying DB fix + integration tests to document the expected behavior

## Related Issues
- closes #891 

## Type of Change
Check all that apply.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Dependency update
- [ ] Refactoring (non-breaking change which improves code quality)
- [ ] Other (please describe)

## Checklist
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

> [!NOTE]
> - [x] I confirm that my Pull Request follows all the checklist requirements above

## Screenshots
If applicable, add screenshots to help explain your changes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved legacy submission exports to preserve scalar, numeric, Boolean, and complex answer values correctly.
  * Corrected validation error details for insufficient JWT signing keys.

* **Test Coverage**
  * Added coverage for scalar and complex answer exports across supported submission scenarios.
  * Added fixture data for comprehensive form submissions, including nested and file-related answers.

* **Development**
  * Updated the development JWT signing key configuration for local use.

### CODE REVIEW

File: tests/Endatix.IntegrationTests/CriticalPaths/Submissions/LegacyExportFormSubmissionsAnswersFlowTests.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        await using AsyncServiceScope scope = world.Services.CreateAsyncScope();
+        ISubmissionExportRepository exportRepository =
+            scope.ServiceProvider.GetRequiredService<ISubmissionExportRepository>();
+
+        List<SubmissionExportRow> rows = [];
+        await foreach (SubmissionExportRow row in exportRepository.GetExportRowsAsync<SubmissionExportRow>(
+                           long.Parse(formId),
+                           sqlFunctionName: "export_form_submissions",
+                           pageSize: 100,
+                           cancellationToken))
```
[the remark this answers is not in the document]
Comment (PeterMarinov) [reply tests/Endatix.IntegrationTests/CriticalPaths/Submissions/LegacyExportFormSubmissionsAnswersFlowTests.cs:77]: Correct finding. This is fixed via adding tenant specific resolution of the sqlFunctionName to ensuretenant scope is verified during the integration testing. Verify

File: src/Endatix.Persistence.SqlServer/Scripts/Procedures/export_form_submissions_v4.sql [hunk 2]
```diff
@@ -121,16 +123,36 @@ BEGIN
...
+                        WHEN 4 THEN JSON_MODIFY(r.AnswersJson, N''' + @jsonPath + N''', JSON_QUERY(o.[value]))
+                        WHEN 5 THEN JSON_MODIFY(r.AnswersJson, N''' + @jsonPath + N''', JSON_QUERY(o.[value]))
+                        ELSE JSON_MODIFY(r.AnswersJson, N''' + @jsonPath + N''', N'''')
+                    END
+                    FROM #Results r
+                    OUTER APPLY (
+                        SELECT j.[type], j.[value]
+                        FROM OPENJSON(r.JsonData) AS j
+                        WHERE j.[key] = N''' + @escapedName + N'''
+                    ) AS o;';
```
[the remark this answers is not in the document]
Comment (PeterMarinov) [reply src/Endatix.Persistence.SqlServer/Scripts/Procedures/export_form_submissions_v4.sql]: Correct. This is fixed. We even went one step further and extracted repeated literals and used variables for cleaner sql structure
